### ISSUE TITLE
Julia doesn't recognize "~" as part of a file name

### ISSUE THREAD

**Issue by ikirill** [emacs-ess/ESS#709]
In this line https://github.com/emacs-ess/ESS/blob/f129e0612ee7d40346b89a0a94393645b85da362/lisp/ess-inf.el#L1275, `abbreviate-file-name` replaces the user's home directory with a `~`, but julia doesn't recognize it as a valid filename, because it doesn't expand them like a shell does.

```
julia> include("~/xxx/f.jl")
+ 
ERROR: could not open file /Users/kirill/xxx/~/xxx/f.jl
```

That change was made in July, and I've found no issue for this, so please tell me if maybe I'm missing something or using ess wrong.

**Comment by jabranham** [comment]
@lionel- related to #586. I guess we'll need to wrap the changes made as a part of that in a wrapper that checks whether we're in julia mode or not and only abbreviate if not?

**Comment by lionel-** [comment]
Perhaps add a `ess--abbreviate-paths` local variable to cust-alists?

**Comment by jabranham** [comment]
I was thinking simpler. Just a wrapper function like:

(defun ess-maybe-abbreviate-path (filename)
  "Abbreviate path unless in julia."
  (if (string= ess-dialect "julia")
    filename
    (abbreviate-file-name path)))

**Comment by lionel-** [comment]
I think we're now trying to avoid this kind of code in ESS. The variable can be `nil` by default and then it only needs to be added to the R cust-alist.

**Comment by jabranham** [comment]
Won't we have to check that variable each time we use `abbreviate-file-name`?

**Comment by lionel-** [comment]
We can do the check in `ess--maybe-abbreviate-path`.

**Comment by jabranham** [comment]
Ah, you're saying just replace the (string= ess-dialect) check with just checking some variable. Makes sense to me.

**Comment by vspinu** [comment]
> ess--abbreviate-paths

How about something more suggestive `ess-dialect-can-abbreviate-paths` and have `ess-abbreviate-file-name` wrapper around `abbreviate-file-name`?

**Comment by lionel-** [comment]
I like it.

**Comment by gdkrmr** [comment]
you need to manually expand the path in julia:
```julia
expanduser("~/Documents")
```

**Comment by jabranham** [comment]
@ikirill can you give an example where this bug is triggered? I think it might have been fixed already by #701

**Comment by ikirill** [comment]
@jabranham You're right. I must have done something wrong, that pr didn't show up for me when I searched for previous issues.
